mitodl/mit-open
# Refactor platform/offered_by filtering and serialization for search

# What are the relevant tickets?
Closes #237 
Closes #113 
Partially addresses #199

# Description (What does it do?)
- Renames `LearningResourcePlatform.platform` to `LearningResourcePlatform.code`
- Refactors serialization and index schema for platform, offered_by
- Adjusts db api and search api to both use `code` property when filtering
- Adds/updates api schema documentation for the above fields.

# How can this be tested?
Run `docker compose up`, some db migrations will run
Run `./manage.py recreate_index --all`

If you haven't already, run the following mgmt commands:
```
./manage.py backpopulate_prolearn_data
./manage.py backpopulate_mitxonline_data
```

Try the following URL's they should all work and filter by the specified offeror/platform correctly.  The total number of results should be the same, though the order will likely differ unless you add a sortby option.

http://localhost:8063/api/v1/learning_resources/?platform=mitxonline
http://localhost:8063/api/v1/learning_resources_search/?platform=mitxonline

http://localhost:8063/api/v1/learning_resources/?offered_by=mitpe
http://localhost:8063/api/v1/learning_resources_search/?offered_by=mitpe


Check the schema documentation, the codes and values for platform, offered_by, department, and resource type should be documented for both the db and opensearch endpoints.

http://localhost:8063/api/v1/schema/redoc/#tag/learning_resources
http://localhost:8063/api/v1/schema/redoc/#tag/learning_resources_search
